src/main/java/org/embulk/output/elasticsearch/ElasticsearchOutputPluginDelegate.java

```diff
@@ -0,0 +1,227 @@
+        @ConfigDefault("7")
+        int getMaximumRetries();
+
+        @Config("initial_retry_interval_millis")
+        @ConfigDefault("1000")
+        int getInitialRetryIntervalMillis();
+
+        @Config("maximum_retry_interval_millis")
+        @ConfigDefault("120000")
+        int getMaximumRetryIntervalMillis();
+
+        @Config("timeout_millis")
+        @ConfigDefault("60000")
+        int getTimeoutMills();
+    }
+
+    public enum Mode
+    {
+        INSERT,
+        REPLACE;
+
+        @JsonValue
+        @Override
+        public String toString()
+        {
+            return name().toLowerCase(Locale.ENGLISH);
+        }
+
+        @JsonCreator
+        public static Mode fromString(String value)
+        {
+            switch (value) {
+                case "insert":
+                    return INSERT;
+                case "replace":
+                    return REPLACE;
+                default:
+                    throw new ConfigException(String.format("Unknown mode '%s'. Supported modes are insert, truncate_insert, replace", value));
+            }
+        }
+    }
+
+    @Override  // Overridden from |OutputTaskValidatable|
+    public void validateOutputTask(PluginTask task, Schema embulkSchema, int taskCount)
+    {
+        if (task.getNodes().size() > 0) {
+            for (NodeAddressTask node : task.getNodes()) {
+                if (node.getHost().endsWith("es.amazonaws.com")) {
+                    throw new ConfigException("This plugin does't support AWS Elasticsearch Service.");
+                }
+            }
+        }
+
+        try (Jetty92RetryHelper retryHelper = createRetryHelper(task)) {
+            log.info(String.format("Connecting to Elasticsearch version:%s", client.getEsVersion(task, retryHelper)));
+
+            if (task.getMode().equals(Mode.REPLACE)) {
+                task.setAlias(Optional.of(task.getIndex()));
+                task.setIndex(client.generateNewIndexName(task.getIndex()));
+                if (client.isExistsIndex(task.getAlias().orNull(), task, retryHelper) && !client.isExistsAlias(task.getAlias().orNull(), task, retryHelper)) {
+                    throw new ConfigException(String.format("Invalid alias name [%s], an index exists with the same name as the alias", task.getAlias().orNull()));
+                }
+            }
+            log.info(String.format("Inserting data into index[%s]", task.getIndex()));
+        }
+    }
+
+    @Override  // Overridden from |ServiceRequestMapperBuildable|
+    public JacksonServiceRequestMapper buildServiceRequestMapper(PluginTask task)
+    {
+        return JacksonServiceRequestMapper.builder()
+                .add(new JacksonAllInObjectScope(), new JacksonTopLevelValueLocator("record"))
+                .build();
+    }
+
+    @Override  // Overridden from |RecordBufferBuildable|
+    public RecordBuffer buildRecordBuffer(PluginTask task)
+    {
+        return new JacksonTaskReportRecordBuffer("records");
```

Comment:
HubSpot output plugin only support bulk insert with target `contact_properties`. In this case, we can avoid OOM by the method separate data into small groups that appropriate with HubSpot request's limitation. 

https://github.com/treasure-data/embulk-output-hubspot/blob/a8b9047ee197121fab697e8842d87125bb7010ee/src/main/java/org/embulk/output/hubspot/HubSpotOutputPluginDelegate.java#L294